### ISSUE TITLE
Change/replace a file for a File Use within a Set

### ISSUE THREAD

**Issue by tmiles2** [emory-libraries/dlp-curate#324]

**Comment by rotated8** [comment]
@eporter23 Does this include any versioning? (I assume not).

**Comment by eporter23** [comment]
No, unless Hyrax is doing that already (as it currently does with file management). Devanshu and I also noted that there seems to be some file versioning stored in Fedora when you change object visibility (oddly enough).

**Comment by devanshu-m** [comment]
@eporter23 when we say change/replace an existing file in the fileset, are we talking about versioning or is that separate that we will talk about in the future? I am not exactly sure what it means to change/replace.

**Comment by devanshu-m** [comment]
Notes:
* Starting to first work on creating a new tab in the fileset edit form that shows the existing files with options to upload files (maybe change or replace? not sure on that yet). Looking at the mockup provided in this ticket.
* Might need an update action in the uploads_controller or re-use the create action
* Hoping to create a new db record with new files when updating files in a fileset
* Need to think about what happens to existing files if they are replaced since we have a one-to-one association between file_type and fileset, meaning, there can be one and only one pmf for every fileset and so. Do we delete the existing file and replace with the new one?

**Comment by eporter23** [comment]
More notes after discussing with Devanshu:
-  For preservation master files, if we need to make a change to that file, that should be treated as a version (replacement, not a deletion and creation). The PMF should not be deleted. Also, Versioning (with a capital v) will be a much larger feature in v.2+ of Curate, where we may need to version the entire work. If we can leverage existing file-level versioning in Hyrax for v.1 that will be a good start.
- For non-master files, as a staff depositor, I may have accidentally assigned the wrong file to a use (e.g. page002.pos as Extracted Text on the page 3 FileSet), so I need an interface to correct that.
- For non-master files, as a staff depositor, I may have accidentally assigned the wrong use to a file (e.g. put page002.pos to Service File where it should have been Extracted Text) so I need an interface to correct that.
- As a staff depositor, I may need to add another file to the FileSet for a use that has not previously been populated (e.g. adding a Service File to an existing FileSet, which has never had a file for that use).

**Comment by devanshu-m** [comment]
After discussion with the SE team and Emily:
* I am going to try and implement the replace strategy (delete and add)

**Comment by eporter23** [comment]
Discuss other options: would it be more straightforward to delete and recreate the FileSet?
